AppCenterIdentity/AppCenterIdentity/MSIdentity.m

```diff
@@ -131,26 +144,132 @@ + (void)handleUrlResponse:(NSURL *)url {
   [MSALPublicClientApplication handleMSALResponse:url];
 }
 
-- (void)acquireToken {
-  if (self.clientApplication == nil) {
-    return;
++ (void)login {
+
+  // TODO protect with canBeUsed.
+  [[MSIdentity sharedInstance] login];
+}
+
+- (void)login {
+  @synchronized(lock) {
+    if (self.clientApplication == nil && self.identityConfig == nil) {
+      self.loginDelayed = YES;
+      return;
+    }
+    self.loginDelayed = NO;
+    [self.clientApplication acquireTokenForScopes:@[ (NSString * _Nonnull) self.identityConfig.identityScope ]
+                                  completionBlock:^(MSALResult *result, NSError *e) {
+                                    // TODO: Implement error handling.
+                                    if (e) {
+                                    } else {
+                                      NSString __unused *accountIdentifier = result.account.homeAccountId.identifier;
+                                      self.accessToken = result.accessToken;
```

Comment:
Should this line be inside a lock? (Just wondering, since there is a lock around the whole method but not inside the completion block. But I generally don't like locks inside callbacks.)

Reply:
This is not part of the implementation. We don't know how we are going to use accessToken in the service, it is hard to determine to lock the handler or not at the moment.
`TODO: synchronize accessToken assignment if it has threading issues` would be enough?

Reply:
Okay that's fine, a TODO would be fine but I'd say not necessary